### PR TITLE
Remove lib directory and replace it with granular utils_* subworkflows

### PR DESCRIPTION
**Pull request by drpatelh**
Some things left to do that I took note of and might be worth splitting up into separate issues:
- [x] Use MultiQC for software versions and remove `custom/dumpsoftwareversions`
- [x] Remove `input_check.nf`, `check_samplesheet.py` and `samplesheet_check.nf` and replace with nf-validation
- [x] Attach MultiQC report in  completion email. This has been removed in the current implementation in this PR.
- [x] Add in methods description implementation. This has been removed in the current implementation in this PR.

Edit by @mashehu:
Not in the scope of this PR: 
- [ ] Move ALL subworkflows and workflows into their own named folder. See [here](https://github.com/nf-core/fetchngs/blob/32952832bae7864a443d9d0acb3f552401987400/modules/local/sra_ids_to_runinfo/main.nf#L2)
- [ ] Remove `modules.config` and have individual `nextflow.config` file in component folders instead. See [here](https://github.com/nf-core/fetchngs/blob/32952832bae7864a443d9d0acb3f552401987400/modules/local/sra_ids_to_runinfo/nextflow.config#L1-L8)
- [ ] Do not remove `nextflow.config` when updating modules/subworkflows
- [ ] Update Github Actions for cloud tests on AWS and Azure. See [here](https://github.com/nf-core/fetchngs/blob/master/.github/workflows/cloud_tests_small.yml) and [here](https://github.com/nf-core/fetchngs/blob/master/.github/workflows/cloud_tests_full.yml)

### CODE REVIEW

**Comment by drpatelh** [comment]
@mirpedrol I added a `checkProfileProvided` function in https://github.com/nf-core/tools/pull/2736/commits/52149765512cc0a5c2c749f82fc1656536a4e4d4

We will need to update the relevant subworkflow in nf-core/modules.

**Comment by drpatelh** [comment]
@mashehu @mirpedrol not sure whether you plan to or are doing this already but I thought we could use nf-test to check the pipeline tests pass as part of the template tests. We just run every nf-test found from the root folder downwards. Then we just add different combinations of nf-core create with different components and run nf-test. Would like to have a quick chat with you both next week if you have some time?

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 1]
```diff
@@ -1,137 +1,72 @@
...
-    }
-}
 
-workflow.onError {
-    if (workflow.errorReport.contains("Process requirement exceeds available memory")) {
-        println("🛑 Default resources exceed availability 🛑 ")
-        println("💡 See here on how to configure pipeline: https://nf-co.re/docs/usage/configuration#tuning-workflow-resources 💡")
-    }
+    emit:
+    multiqc_report = MULTIQC.out.report // channel: /path/to/multiqc_report.html
```
Comment (mirpedrol) [inline nf_core/pipeline-template/workflows/pipeline.nf:68]:
```suggestion
    multiqc_report = MULTIQC.out.report.toList() // channel: /path/to/multiqc_report.html
```

**Review by mirpedrol — APPROVED** [review]

**Review by mashehu — APPROVED** [review]
Still need to run a few test runs locally, but I think it looks good

File: nf_core/lint/files_exist.py [hunk 2]
```diff
@@ -200,7 +197,14 @@ def files_exist(self) -> Dict[str, Union[List[str], bool]]:
         Path("lib", "Completion.groovy"),
         Path("lib", "Workflow.groovy"),
     ]
-    files_warn_ifexists = [Path(".travis.yml")]
+    files_warn_ifexists = [
+        Path(".travis.yml"),
+        Path("lib", "NfcoreTemplate.groovy"),
```
Comment (ewels) [inline nf_core/lint/files_exist.py:202]: Duplicated

File: nf_core/lint/files_exist.py [hunk 3]
```diff
@@ -200,7 +197,14 @@ def files_exist(self) -> Dict[str, Union[List[str], bool]]:
...
+        Path("lib", "Utils.groovy"),
+        Path("lib", "WorkflowMain.groovy"),
+        Path("lib", "NfcoreTemplate.groovy"),
+        Path("lib", f"Workflow{short_name[0].upper()}{short_name[1:]}.groovy"),
```
Comment (ewels) [inline nf_core/lint/files_exist.py:206]: Switch these to fails.

**Review by mashehu — APPROVED** [review]

**Pull request merged**